devenv/local/docker-compose/tests/devnet-liveness.sh

```diff
@@ -0,0 +1,261 @@
+if [[ $NAKAMOTO_SIGNER_DOCKER_LOGS == *"Signer spawned successfully"* ]]; then
+    NAKAMOTO_SIGNER_READY_SUCCESS=true
+    echo "Nakamoto Signer || Signer spawned successfully"
+    NAKAMOTO_SIGNER_READY_SUCCESS_FRMT=$(echo "\033[1;32m$NAKAMOTO_SIGNER_READY_SUCCESS\033[0m ✅")
+fi
+
+
+echo "\033[1mNAKAMOTO_SIGNER_READY_SUCCESS\033[0m: $NAKAMOTO_SIGNER_READY_SUCCESS_FRMT"
+echo "\n"
+
+
+
+
+
+
+
+
+echo " --------------------------------------------------"
+echo "| => (6) 🔬 TEST: [CHECK IF STACKS NODE IS READY]  |"
+echo " --------------------------------------------------"
+
+
+GET_STACKS_NODE_INFO_STATUS_CODE=$(curl --write-out %{http_code} --silent --output /dev/null "http://localhost:20443/v2/info")
+
+
+echo "\nGET STACKS NODE STATUS: $GET_STACKS_NODE_INFO_STATUS_CODE"
+
+STX_LIVENESS_SUCCESS=false 
+STACKS_LIVENESS_SUCCESS_FRMT=$(echo "\033[1;31m$STX_LIVENESS_SUCCESS\033[0m❌")
+
+if [[ $GET_STACKS_NODE_INFO_STATUS_CODE == "200" ]]; then
+    STX_LIVENESS_SUCCESS=true
+    STACKS_LIVENESS_SUCCESS_FRMT=$(echo "\033[1;32m$STX_LIVENESS_SUCCESS\033[0m ✅")
+fi
+
+
+echo "\033[1mSTACKS_LIVENESS_SUCCESS\033[0m: $STACKS_LIVENESS_SUCCESS_FRMT"
+echo "\n"
+
+
+
+
+
+
+
+
+echo " ---------------------------------------------------------------"
+echo "| => (7) 🔬 TEST: [CHECK IF STX NODE IS SYNCED WITH BTC UTXOs]  |"
+echo " ---------------------------------------------------------------"
+
+
+## (RPC APPROACH)
+# GET_STACKS_NODE_INFO=$(curl -s "http://localhost:20443/v2/info")
+
+# echo "\nGET STACKS NODE INFO:"
+# echo $GET_STACKS_NODE_INFO | jq 'del(.stackerdbs)'
+# echo "\t\t.\n\t\t.\n  \033[1;32m<<\033[0m \033[1;35mLong Output Supressed\033[0m \033[1;32m>>\033[0m \n\t\t.\n\t\t."
+
+# STX_SYNC_WITH_BTC_UTXO_SUCCESS=$(echo $GET_STACKS_NODE_INFO | jq -r '.stacks_tip_height != 0')
+# STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT=$([ "$STX_SYNC_WITH_BTC_UTXO_SUCCESS" == "true" ] && echo "\033[1;32m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m ✅" || echo "\033[1;31m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m❌") 
+
+# echo "\033[1mSTX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m: $STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT"
+# echo "\n"
+
+
+## (LOGS APPROACH)
+STACKS_DOCKER_LOGS=$(docker logs stacks 2>/dev/null)
+
+
+STX_SYNC_WITH_BTC_UTXO_SUCCESS=false
+STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT=$(echo "\033[1;31m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m❌")
+if [[ $STACKS_DOCKER_LOGS == *"UTXOs found"* ]]; then
+    STX_SYNC_WITH_BTC_UTXO_SUCCESS=true
+    echo "Stacks || UTXOs found - will run as a Miner node"
+    STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT=$(echo "\033[1;32m$STX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m ✅")
+fi
+
+echo "\033[1mSTX_SYNC_WITH_BTC_UTXO_SUCCESS\033[0m: $STX_SYNC_WITH_BTC_UTXO_SUCCESS_FRMT"
+echo "\n"
+
+
+
+
+echo " ---------------------------------------------------------------"
+echo "| => (8) 🔬 TEST: [CHECK STACKS-API LIVENESS]  |"
+echo " ---------------------------------------------------------------"
+
+
+GET_STACKS_API_PING=$(curl -s "http://localhost:3700")
```

Comment:
This is the port for the API's event observer, which is good, but we should also have a liveness check for the "public" API (port 3999)

Reply:
Just added the `STACKS_PUBLIC_API_LIVENESS_SUCCESS` test. Everything passes now